src/msw/dirdlg.cpp

```diff
@@ -212,38 +146,54 @@ void wxDirDialog::SetPath(const wxString& path)
     }
 }
 
+wxString wxDirDialog::GetPath() const
+{
+    wxCHECK_MSG(!HasFlag(wxDD_MULTIPLE), wxEmptyString, "GetPath called when wxDD_MULTIPLE defined");
+
+    return m_path;
+}
+
+void wxDirDialog::GetPaths(wxArrayString& paths) const
+{
+    paths = m_paths;
+}
+
 int wxDirDialog::ShowModal()
 {
     WX_HOOK_MODAL_DIALOG();
 
     wxWindow* const parent = GetParentForModalDialog();
     WXHWND hWndParent = parent ? GetHwndOf(parent) : NULL;
 
+    m_paths.clear();
+
     // Use IFileDialog under new enough Windows, it's more user-friendly.
     int rc;
-#if wxUSE_IFILEDIALOG
+#if wxUSE_IFILEOPENDIALOG
     // While the new dialog is available under Vista, it may return a wrong
     // path there (see http://support.microsoft.com/kb/969885/en-us), so we
     // don't use it there by default. We could improve the version test to
     // allow its use if the comdlg32.dll version is greater than 6.0.6002.22125
     // as this means that the hotfix correcting this bug is installed.
     if ( wxGetWinVersion() > wxWinVersion_Vista )
     {
-        rc = ShowIFileDialog(hWndParent);
+        rc = ShowIFileOpenDialog(hWndParent);
     }
     else
     {
         rc = wxID_NONE;
     }
 
     if ( rc == wxID_NONE )
-#endif // wxUSE_IFILEDIALOG
+#endif // wxUSE_IFILEOPENDIALOG
     {
         rc = ShowSHBrowseForFolder(hWndParent);
     }
 
     // change current working directory if asked so
-    if ( rc == wxID_OK && HasFlag(wxDD_CHANGE_DIR) )
+    // but only if the dialog is single selection only
+    if ( rc == wxID_OK &&
+         ( HasFlag(wxDD_CHANGE_DIR) && !HasFlag(wxDD_MULTIPLE) ) )
```

Comment:
I'm not convinced we should do this. We've already given them an assert in ctor to say that this combination is not allowed. And it doesn't matter what we do when preconditions are not satisfied, so why complicate the code with more checks. Personally, I'd just leave the original code unchanged and rely on `wxDD_CHANGE_DIR` presence implying `wxDD_MULTIPLE` absence.

Reply:
OK, will revert.

I just look at things from the user perspective, where it can mean that the CWD is changed to a seemingly random folder. Surprisingly, some people even do not use the debug versions of libraries for development...

Reply:
They don't have to use debug build of wx for development and they don't have to read the docs, which document that this combination is unsupported, neither, but I don't think we have to go to any special trouble to support them neither...